src/test/java/seedu/address/model/expense/UniqueExpenseListTest.java

```diff
@@ -1,4 +1,175 @@
 package seedu.address.model.expense;
 
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static seedu.address.logic.commands.CommandTestUtil.VALID_DESCRIPTION_BUILD_A_BEAR;
+import static seedu.address.logic.commands.CommandTestUtil.VALID_EXPENSE_CATEGORY_TRANSPORT;
+import static seedu.address.testutil.Assert.assertThrows;
+import static seedu.address.testutil.TypicalExpenses.ANNUAL_NETFLIX_FEES;
+import static seedu.address.testutil.TypicalExpenses.ANNUAL_SPOTIFY;
+
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+
+import org.junit.jupiter.api.Test;
+
+import seedu.address.model.expense.exceptions.DuplicateExpenseException;
+import seedu.address.model.expense.exceptions.ExpenseNotFoundException;
+import seedu.address.testutil.ExpenseBuilder;
+
 public class UniqueExpenseListTest {
+
+    private final UniqueExpenseList UniqueExpenseList = new UniqueExpenseList();
+
+    @Test
+    public void contains_nullExpense_throwsNullPointerException() {
+        assertThrows(NullPointerException.class, () -> UniqueExpenseList.contains(null));
+    }
+
+    @Test
+    public void contains_expenseNotInList_returnsFalse() {
+        assertFalse(UniqueExpenseList.contains(ANNUAL_SPOTIFY));
+    }
+
+    @Test
+    public void contains_expenseInList_returnsTrue() {
+        UniqueExpenseList.add(ANNUAL_SPOTIFY);
+        assertTrue(UniqueExpenseList.contains(ANNUAL_SPOTIFY));
+    }
+
+    // Unique identity fields: all fields but ExpenseCategory
```

Comment:
Can I just confirm that if all fields same but expense category different means the same object (shouldn't be the case right)?

Reply:
In this case, the question of identity is different from equals method right? so in a way, I cannot add 2 items with the same identity, but i essentially still consider them as not equal objects (?)

Reply:
Or should this test case be renamed? It sounds like they have different identity because expense category is different?

Identity for our application's Expense class hasn't been defined yet.

Reply:
yeah this was more of a design choice as I was thinking sometimes you have expenses which may fall under two categories. I was imagining it more in the context of tags. For example if I have a taken a taxi which can be claimed under work I may want to include it in two categories. I was not too sure about this one as well we can change it to expense category being part of the primary key